contracts/CombinedChainlinkOracle.vy

```diff
@@ -0,0 +1,33 @@
+struct LatestRoundData:
+    round_id: uint80
+    answer: int256
+    started: uint256
+    updated: uint256
+    answered_round: uint80
+
+interface ChainlinkOracle:
+    def latestRoundData() -> LatestRoundData: view
+    def decimals() -> uint256: view
+
+SCALE: constant(int256) = 10**18
+
+yfi_oracle: public(immutable(ChainlinkOracle)) # YFI/USD
+eth_oracle: public(immutable(ChainlinkOracle)) # ETH/USD
+decimals: public(constant(uint256)) = 18
+
+@external
+def __init__(_yfi_oracle: address, _eth_oracle: address):
+    yfi_oracle = ChainlinkOracle(_yfi_oracle)
+    eth_oracle = ChainlinkOracle(_eth_oracle)
+    assert ChainlinkOracle(_yfi_oracle).decimals() == 8
+    assert ChainlinkOracle(_eth_oracle).decimals() == 8
+
+@external
+@view
+def latestRoundData() -> LatestRoundData:
+    yfi: LatestRoundData = yfi_oracle.latestRoundData()
```

Comment:
both of these data sources have 3600s heartbeat, im thinking we can check each one individually to be sure theres redundancy in sanity check?

Reply:
Sometimes is recommended to add checks for price deviation, im wondering here since deviation on yfi is 1% and ETH is 0.5% if this could be useful.

e.g

```
if (abs(cl.answer - lastYFIGoodPrice) / lastYFIGoodPrice > YFI_TOLERANCE) {
    step 6; // means chainlink failed the check
} else {
   lastYFIGoodPrice = cl.answer
}
```

Reply:
I think for our usecase we can keep it like this. The potential upside for an attacker is small. And the fact that we use `max(chainlink, curve)` means they would have to manipulate the curve pool simultaneously, making it very unlikely.